fix: fixing failing goals, cache and token tests

ai16z-demirix opened the pull request:
<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:
https://github.com/ai16z/eliza/issues/519
<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section is to be filled out before final review and merge. -->

# Risks
low - fixing tests
<!--
Low, medium, large. List what kind of risks, and what could be effected.
-->

# Background

## What does this PR do?
Fixing goals, cache and token tests
## What kind of change is this?
Bug-fixes
<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there is no linked issue explaining why. If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->
Fixing tests to prevent flaky tests.
# Documentation changes needed?
None
<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If a docs change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested, and probably help the PR get merged quicker. -->

# Testing

## Where should a reviewer start?
navigate to packages/core
## Detailed testing steps
run pnpm install, pnpm test
<!--
None, automtated tests are fine.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deploy, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste commandline output. -->
<!--
## Database changes
-->

<!--  If there is something more than the automated steps, please specifiy deploy instructions. -->
<!--
## Deployment instructions
-->

<!-- If you are on Discord, please join https://discord.gg/ai16z and state your Discord username here for contribute role and join us in #development-feed -->
<!--
## Discord username

-->

pgoos reviewed (changes requested):
Hey @ai16z-demirix . Nice work! Have you actually run the tests before creating the PR? When I run them in codespaces with `npm run test` I am getting a bunch of failures. Some of them are related to `No test files found, exiting with code 1` (we should fix that for CI purposes) but there's also `token.test.ts` test file that fails with:
```
 FAIL  src/tests/token.test.ts [ src/tests/token.test.ts ]
Error: Failed to resolve entry for package "@ai16z/plugin-solana". The package may have incorrect main/module/exports specified in its package.json.
```
The thing is that I see `@ai16z/plugin-solana` installed as a dependency. I think the problem related to imported modules is bigger. For example, if you open `agent/src/index.ts`, you can see that most of the imports are not referenced correctly.

ai16z-demirix commented:
Hello @pgoos thanks for reaching out! Yes, I have run tests locally multiple times before creating the PR. 
You should maybe try runing pnpm test, instead of npm test.
Honestly, I am confused with this error: No test files found, exiting with code 1
Are you sure that you are executing tests in packages/core directory? 
This is my output:
![Screenshot from 2024-11-23 18-10-09](https://github.com/user-attachments/assets/b89fcb68-722a-4daa-a26a-0ce90de99dc1)

The other error with @ai16z/plugin-solana persisted also on my local until I ran pnpm build. We can figure this out, there is problem with imports definitely.

yakul0 commented:
ai16z is full of scammers do not work or contribute code to these people.

They have been pumping and dumping crypto coins for the past week on retail investors: $eliza, $matr, $vs and more. 

https://dexscreener.com/solana/2pofqcvn4wysazpsqppxq4cspbtkqlsehf9q5x77avm7
https://dexscreener.com/solana/hzqcylo7gjl5xvvglc2zfgfuwzkunhrzvge2hn9sysbr

scammers:
https://x.com/mattyryze
https://x.com/shawmakesmagic/status/1857903111073067315
https://x.com/pindarosothecat

Bunch of people calling out the ai16z team: 
https://x.com/spacepixel/status/1860511246241661274
https://x.com/JW100x/status/1860501669907189772
https://x.com/0xApeRich/status/1860492861357060381
https://x.com/BullTrx1/status/1860492092767695108

DYO. Don't help work for free for these people. Fork the repo and contribute somewhere else.

lalalune commented on packages/core/package.json:
```diff
@@ -61,6 +60,7 @@
         "@ai16z/adapter-sqlite": "workspace:*",
         "@ai16z/adapter-sqljs": "workspace:*",
         "@ai16z/adapter-supabase": "workspace:*",
+        "@ai16z/plugin-solana": "workspace:*",
```
We can't have this reference here, this test sould be moved to the solana plugin

lalalune reviewed (commented):
Please fix package.json, move token test to solana plugin

ai16z-demirix commented:
@lalalune Fixed! Thank you for this input. 
Fixed package.json file and moved token.test to solana plugin. Added vitest to that package. Tests are running properly now.
